Tools/TestWebKitAPI/Tests/WebCore/FlatDynamicRangeLimitTests.cpp

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (C) 2016 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+
+#include <WebCore/FlatDynamicRangeLimit.h>
+
+#include <WebCore/StyleDynamicRangeLimit.h>
+
+namespace TestWebKitAPI {
+
+TEST(FlatDynamicRangeLimit, DefaultConstruction)
+{
+    auto def = WebCore::FlatDynamicRangeLimit();
+    EXPECT_EQ(def.value(), 1.0f);
+    EXPECT_EQ(def.applicableValue(), WebCore::ApplicableDynamicRangeLimit::NoLimit);
+
+    static_assert(WebCore::FlatDynamicRangeLimit().value() == 1);
+    static_assert(WebCore::FlatDynamicRangeLimit().applicableValue() == WebCore::ApplicableDynamicRangeLimit::NoLimit);
```

Comment:
I think this can be moved to the file `FlatDynamicRangeLimit.cpp`.

Reply:
Well, I'm *testing* that this function is `constexpr`, so putting that in a test felt more appropriate.
If it was in FlatDynamicRangeLimit.cpp, I would consider it too obvious to keep!